apis/v1alpha1/gateway_types.go

```diff
@@ -77,6 +77,10 @@ type GatewaySpec struct {
 	// As a special case, each group of compatible listeners
 	// may contain exactly one Listener with a match type of "Any".
 	//
+	// Listeners across Gateway resources cannot be collapsed i.e. Gateway
```

Comment:
Doesn't L84 directly contradict with this? Or am I reading this wrong

Reply:
No. L84 refers to listeners within the same Gateway resource.

Reply:
On L67, L84, and L93, should "GatewayClass" be changed to "Gateway" or perhaps "implementation"?  You are saying the listeners within the _Gateway_ can be collapsed, so it seems odd to say that the _GatewayClass_ collapses the listeners.